tools/link.py

```diff
@@ -514,6 +514,7 @@ def setup_pthreads():
   settings.REQUIRED_EXPORTS += [
     '_emscripten_thread_free_data',
     '_emscripten_thread_crashed',
+    '_emscripten_proxy_promise_finish',
```

Comment:
I don't think this should be needed here.  At least I don't see runtime_pthread.js depending on this.  Instead I think you can add this to `_emscripten_receive_on_main_thread_js__deps`